src/Symfony/Component/HttpFoundation/Tests/RequestTest.php

```diff
@@ -326,6 +326,20 @@ public function testGetMimeTypeFromFormat($format, $mimeTypes)
         }
     }
 
+    /**
+     * @dataProvider getFormatToMimeTypeMapProviderForAccept
+     */
+    public function testGetMimeTypeFromFormatWithAccept($format, $mimeTypes, $accepts)
+    {
+        if (null !== $format) {
+            $request = new Request();
+            $request->headers->set('Accept', $accepts);
+            $request->setFormat($format, $mimeTypes);
+            //Only when accept value matches a value verbatim inside of the mimetype array.
+            $this->assertEquals($accepts, $request->getMimeType($format));
```

Comment:
Yes, the test looks better, but my test still doesn't pass isn't it? I don't know if some people attempt to not get the mime type based on `Accept` header but I think this is a serious behavior change that may break some code built on top of it (as suggest my test).